pkg/auth/oauth2/types.go

```diff
@@ -11,3 +11,11 @@ type Config struct {
 	UserInfoURL    string
 	RedirectPath   string // RedirectPath is the path that the user will be redirected to after login
 }
+
+type Provider string
+
+const (
+	Authentik Provider = "authentik"
+	Google    Provider = "Google"
+	GitHub    Provider = "GitHub"
```

Comment:
the provider case should be consistent.
declaire `google` and `github` here, and always convert the provider to lowercase when using it